src/cpp/src/llm_pipeline.cpp

```diff
@@ -247,57 +237,63 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
+                size_t seq_length = m_tokenized_chat_history.input_ids.get_shape().at(1);
+
+                utils::initialize_position_ids(*position_ids, concatenated_attention_mask, seq_length, start_pos);
+            } else {
+                position_ids = ov::Tensor{ov::element::i64, input_ids.get_shape()};
+                utils::initialize_position_ids(*position_ids, attention_mask, attention_mask.get_shape()[1], kv_cache_len);
+            }
         }
 
         if(m_adapter_controller) {
             m_adapter_controller->apply(m_model_runner, config.adapters);
         }
 
         ov::genai::EncodedResults result;
-        if (config.is_beam_search() && is_chat_conversation) {
-            std::tie(result, m_selected_beam) = beam_search(m_model_runner, input_ids, concatenated_attention_mask,
-                                                            config, position_ids, m_selected_beam);
-        } else {
-            std::vector<SequenceGroup::Ptr> requests;
-            size_t block_size = 1;
-            bool enable_prefix_caching = false;
-
-            config.stop_token_ids.insert(config.eos_token_id);
-            for (size_t request_id = 0; request_id < batch_size; request_id++) {
-                SequenceGroup::Ptr sequence_group;
-                if (is_chat_conversation && !m_is_cache_empty) {
-                    sequence_group = std::make_shared<SequenceGroup>(request_id, m_tokenized_chat_history.input_ids, config, block_size, enable_prefix_caching);
-                    sequence_group->update_processed_tokens_num(m_tokenized_chat_history.input_ids.get_shape().at(1) - 1);
-                } else {
-                    size_t seq_len = input_ids.get_shape().at(1);
-                    size_t batch_offset = request_id * seq_len;
-                    const int64_t* prompt_start = input_ids.data<const int64_t>() + batch_offset;
-                    std::vector<int64_t> tokenized_prompt(prompt_start, prompt_start + seq_len);
-                    // in case of multi batch scenario, remove eos_token_id at start of prompt
-                    auto real_prompt_start = std::find_if(tokenized_prompt.begin(), tokenized_prompt.end(), [&config](int64_t token) { return token != config.eos_token_id; });
-                    tokenized_prompt.erase(tokenized_prompt.begin(), real_prompt_start);
-
-                    sequence_group = std::make_shared<SequenceGroup>(request_id, tokenized_prompt, config, block_size, enable_prefix_caching);
-                    sequence_group->update_processed_tokens_num(tokenized_prompt.size() - 1);
-                }
+        std::vector<SequenceGroup::Ptr> requests;
+        size_t block_size = 1;
+        bool enable_prefix_caching = false;
 
-                sequence_group->set_sequence_group_ptr(sequence_group);
-                requests.push_back(sequence_group);
+        config.stop_token_ids.insert(config.eos_token_id);
```

Comment:
why do we need it? Looks like some WA for generation config

is not it fixed by https://github.com/openvinotoolkit/openvino.genai/pull/1196/files#diff-0e465efbb47cb89358103505d30529f860a4356496843ae50075ea8788ad5d72R198 where EOS token is always added to `stop_token_ids` via `config.set_eos_token_id` ?

Reply:
yes, it works without that line, i didn't notice, thanks!
removed